openedx/features/pakx/lms/pakx_admin_app/views.py

```diff
@@ -0,0 +1,132 @@
+            return Response(self.serializer_class(self.queryset.order_by('-id'), many=True).data)
+
+        else:
+            queryset = None
+
+            if self.request.query_params.get('roles', None) is not None:
+                roles = self.request.query_params.get('roles').split(',')
+                queryset = self.filter_users_by_roles(roles, queryset)
+
+            if self.request.query_params.get('languages', None) is not None:
+                if self.request.query_params.get('roles', None) is None or queryset is not None:
+                    languages = self.request.query_params.get('languages').split(',')
+                    queryset = self.filter_users_by_languages(languages, queryset)
+
+            if self.request.query_params.get('sort_by', None) is not None:
+                sort_by = self.request.query_params.get('sort_by').split(',')[
+                    len(self.request.query_params.get('sort_by').split(',')) - 1]
+                sort_by = '-' + sort_by
+
+            else:
+                sort_by = '-id'
+
+            if queryset is None:
+                queryset = User.objects.filter(pk=0)
+
+            return Response(self.serializer_class(queryset.order_by(sort_by), many=True).data)
+
+    def filter_users_by_roles(self, roles, queryset):
+        for role in roles:
+            if self.is_superuser_request(role):
+                queryset = self.get_superusers(queryset)
+            if self.is_staff_request(role):
+                queryset = self.get_staff(queryset)
+            if self.is_training_manager_request(role):
+                queryset = self.get_training_managers(queryset)
+
+        return queryset
+
+    def filter_users_by_languages(self, languages, queryset):
+        if queryset is None:
+            return User.objects.filter(profile__language__in=languages)
+        else:
+            return queryset.filter(profile__language__in=languages)
+
+    def is_superuser_request(self, role):
+        # 1 represents superuser/admin
+        return role == '1'
+
+    def is_staff_request(self, role):
+        # 2 represents staff
+        return role == '2'
+
+    def is_training_manager_request(self, role):
+        # 3 represents training_manager
+        return role == '3'
+
+    def get_superusers(self, queryset):
+        if queryset is None:
+            return User.objects.filter(is_superuser=True)
+        else:
+            return (queryset | User.objects.filter(is_superuser=True)).distinct()
+
+    def get_staff(self, queryset):
+        if queryset is None:
+            return User.objects.filter(is_staff=True)
+        else:
+            return (queryset | User.objects.filter(is_staff=True)).distinct()
+
+    def get_training_managers(self, queryset):
+        if queryset is None:
+            return User.objects.filter(groups__name=GROUP_TRAINING_MANAGERS)
+        else:
+            return (queryset | User.objects.filter(groups__name=GROUP_TRAINING_MANAGERS)).distinct()
```

Comment:
the Django for this is managers, https://www.django-rest-framework.org/api-guide/filtering/#djangofilterbackend